src/inference/graph_inference.cpp

```diff
@@ -1053,12 +1256,24 @@ void GraphInference::SaveModel(const std::string& file_prefix) {
   LOG(INFO) << "Saving model " << file_prefix << "...";
   FILE* ffile = fopen(StringPrintf("%s_features", file_prefix.c_str()).c_str(), "wb");
   int num_features = features_.size();
+  int num_factor_features = factor_features_.size();
   fwrite(&num_features, sizeof(int), 1, ffile);
+  fwrite(&num_factor_features, sizeof(int), 1, ffile);
```

Comment:
I noticed this problem that your binary format is not incompatible with the old binary format. Could you make it compatible? Put the number of factor features after the regular features and when reading, skip factors if we are the end of the file.